service/controller/machine_deployment_drainer_resource_set.go

```diff
@@ -168,3 +175,12 @@ func newMachineDeploymentDrainerResourceSet(config machineDeploymentDrainerResou
 
 	return resourceSet, nil
 }
+
+func machineDeploymentDrainerLabelSelectorFunc(cr metav1.Object) *metav1.LabelSelector {
+	return &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			label.Cluster:           key.ClusterID(cr),
+			label.MachineDeployment: key.MachineDeploymentID(cr),
+		},
+	}
+}
```

Comment:
With this we make the label selector configurable and use the same across the `drainer` and `drainerfinisher` resource.